commit 56363dff436c12eaa296bbba39dfd65ec3f1f1f2
Author: chrfranke <authors@smartmontools.org>
Date:   Sat Nov 6 15:07:18 2021 +0000

    update-smart-drivedb.in: Add '-q' option to suppress info messages.
    (GH issues/110).
    update-smart-drivedb.8.in: Document new option.
    
    git-svn-id: http://svn.code.sf.net/p/smartmontools/code/trunk@5242 4ea69e1a-61f1-4043-bf83-b5c94c648137

--- a/update-smart-drivedb.8.in
+++ b/update-smart-drivedb.8.in
@@ -1,6 +1,6 @@
 .ig
 Copyright (C) 2013 Hannes von Haugwitz <hannes@vonhaugwitz.com>
-Copyright (C) 2014-20 Christian Franke
+Copyright (C) 2014-21 Christian Franke
 
 SPDX-License-Identifier: GPL-2.0-or-later
 
@@ -144,6 +144,10 @@ Print the OpenPGP/GPG public key block.
 .B \-\-dryrun
 Print download commands only.
 .TP
+.B \-q
+[NEW EXPERIMENTAL UPDATE-SMART-DRIVEDB FEATURE]
+Suppress info messages.
+.TP
 .B \-v
 Verbose output.
 .Sp
--- a/update-smart-drivedb.in
+++ b/update-smart-drivedb.in
@@ -72,6 +72,7 @@ Usage: $myname [OPTIONS] [DESTFILE]
   --no-verify     Don't verify signature
   --export-key    Print the OpenPGP/GPG public key block
   --dryrun        Print download commands only
+  -q              Suppress info messages
   -v              Verbose output
 
 Updates $DRIVEDB
@@ -131,6 +132,11 @@ inpath()
   return $rc
 }
 
+iecho()
+{
+  test -n "$quiet" || echo "$*"
+}
+
 vecho()
 {
   test -n "$q" || echo "$*"
@@ -522,6 +528,7 @@ mv_all()
 smtctl=$SMARTCTL
 tool=
 urlid="svn"
+quiet=
 q="-q"
 dryrun=
 trunk=
@@ -547,6 +554,9 @@ while true; do case $1 in
     shift; test -n "$1" || usage
     urlid=$1 ;;
 
+  -q)
+    quiet=t ;;
+
   -v)
     q= ;;
 
@@ -727,7 +737,7 @@ if [ -f "$DEST" ]; then
     then
       rm -f "$DEST.new" "$DEST.new.raw" "$DEST.new.raw.asc"
       touch "$DEST.lastcheck"
-      echo "$DEST is already up to date"
+      iecho "$DEST is already up to date"
       exit 0
     fi
     mv_all "$DEST" "" ".old"
@@ -738,4 +748,4 @@ fi
 
 mv_all "$DEST" ".new" ""
 
-echo "$DEST updated from ${trunk:-branches/$brnch}${no_verify:+ (NOT VERIFIED)}"
+iecho "$DEST updated from ${trunk:-branches/$brnch}${no_verify:+ (NOT VERIFIED)}"
